### PR TITLE
Slightly Cleaner Mess!

### PR DESCRIPTION
**Pull request by NegativeScript**

### CODE REVIEW

File: furusiya/function_mess.py [hunk 1]
```diff
@@ -202,25 +241,31 @@ def new_game():
     Game.player.gain_xp(40 + 80 + 160 + 320)
 
 
+def mini_loop(condition, callback):
```
Comment (nightblade9) [inline furusiya/function_mess.py]: Probably needs a better name. I don't have one in mind.

File: furusiya/main_interface.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ class Game:
     area_map = None
     renderer = None
     ui = None
+    turn = None
```
Comment (nightblade9) [inline furusiya/main_interface.py]: I would prefer this to be called `current_turn` or `whose_turn`

File: furusiya/function_mess.py [hunk 3]
```diff
@@ -39,6 +39,17 @@ def msgbox(text, width=50):
...
 def handle_keys():
+    if (Game.player.turns_to_rest > 0
+            and not [
+                e
+                for e in Game.area_map.entities
+                if e.hostile and (e.x, e.y) in Game.renderer.visible_tiles
+            ]):
+        Game.player.turns_to_rest -= 1
+        Game.player.rest()
+        Game.turn = None
```
Comment (nightblade9) [inline furusiya/function_mess.py]: This makes me cringe. Who's turn is `None`? It's not obvious. (Scrollin down, it's the other AIs turns...)

File: furusiya/model/entities/party/player.py [hunk 4]
```diff
@@ -48,14 +51,20 @@ def unmount(self, horse):
             self.mounted = False
             horse.is_mounted = False
 
+    def _rest(self, max_hp):
```
Comment (nightblade9) [inline furusiya/model/entities/party/player.py]: Please rename to `_get_health_for_resting` or something equally verbose.

File: furusiya/model/entities/party/player.py [hunk 5]
```diff
@@ -71,7 +80,12 @@ def move_or_attack(self, dx, dy):
         else:
             self.move(dx, dy)
             if self.mounted:
-                self.move(dx, dy)
+                if self.mounted_moves >= 1:
```
Comment (nightblade9) [inline furusiya/model/entities/party/player.py]: I don't understand what these counts are for. Maybe more verbose names will increase understandability?

File: furusiya/model/systems/ai_system.py [hunk 6]
```diff
@@ -21,3 +16,9 @@ def get_ai(cls, owner):
     @classmethod
     def has_ai(cls, owner):
         return cls.get_ai(owner) is not None
+
+    @classmethod
```
Comment (nightblade9) [inline furusiya/model/systems/ai_system.py]: Can this be a `staticmethod` instead? If `cls` is always the same, we can just reference it directly (eg. `GameObject.get_ai`).

Comment (NegativeScript) [reply furusiya/model/systems/ai_system.py]: Probably. A `classmethod` works just fine as well. I don't see a reason to change it.

**Pull request merged**